tensorboard/pip_package/build_pip_package.sh

```diff
@@ -12,66 +12,92 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-# ==============================================================================
-
-
 set -e
+if [ -z "${RUNFILES}" ]; then
+  RUNFILES="$(CDPATH= cd -- "$0.runfiles" && pwd)"
+fi
 
-die() {
-  printf >&2 '%s\n' "$1"
-  exit 1
+if [ "$(uname)" = "Darwin" ]; then
+  sedi="sed -i ''"
+else
+  sedi="sed -i"
+fi
+
+smoke() {
+  virtualenv -qp python$1 venv$1
+  cd venv$1
+  . bin/activate
+  pip install -qU pip
+  pip install -qU tf-nightly
+  pip install -qU ../dist/*py$1*.whl >/dev/null
+  mkdir logs
+  mkfifo pipe
+  tensorboard --port=0 --logdir=logs 2>pipe &
+  perl -ne 'print STDERR;/http:.*:(\d+)/ and print $1.v10 and exit 0' <pipe >port
```

Comment:
I'm not sure if it's possible to do with `$()` syntax. We are in a tmp directory, this doesn't overflow 80 column, and is readable. We'd need to sleep() poll a `/tmp/tbport.$PID` file. I prefer the POSIX FIFO since it's faster.